src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -325,18 +323,26 @@ private async Task UnsynchronizedStartHostAsync(ScriptHostStartupOperation activ
                 else
                 {
                     logger.LogDebug($"Will start a new host after delay.");
-                    await Utility.DelayWithBackoffAsync(attemptCount, currentCancellationToken, min: TimeSpan.FromSeconds(1), max: TimeSpan.FromMinutes(2), logger: logger)
-                        .ContinueWith(t =>
-                        {
-                            if (currentCancellationToken.IsCancellationRequested)
-                            {
-                                logger.LogDebug($"Cancellation for operation '{activeOperation.Id}' requested during delay. A new host will not be started.");
-                                currentCancellationToken.ThrowIfCancellationRequested();
-                            }
-
-                            logger.LogDebug("Starting new host after delay.");
-                            return StartHostAsync(currentCancellationToken, attemptCount, parentOperationId: activeOperation.Id);
-                        });
+
+                    try
+                    {
+                        await Utility.DelayWithBackoffAsync(attemptCount, currentCancellationToken, min: TimeSpan.FromSeconds(1), max: TimeSpan.FromMinutes(2), logger: logger);
+                    }
+                    catch (Exception ex)
+                    {
+                        var delay = TimeSpan.FromSeconds(5);
```

Comment:
Perhaps this special error handling should be moved into Utility.DelayWithBackoffAsync so it'll apply everywhere?

Reply:
Yeah I was thinking the same thing; I'll do that. It's really just a catch-all in case there's some other bug I didn't find during this. But feels better to be safe.